gtk4/src/subclass/entry_buffer.rs

```diff
@@ -0,0 +1,242 @@
+            if let Some(f) = (*parent_class).inserted_text {
+                f(
+                    entry_buffer
+                        .unsafe_cast_ref::<EntryBuffer>()
+                        .to_glib_none()
+                        .0,
+                    position,
+                    chars.to_glib_none().0,
+                    chars.len() as u32,
+                )
+            }
+        }
+    }
+}
+
+unsafe impl<T: EntryBufferImpl> IsSubclassable<T> for EntryBuffer {
+    fn override_vfuncs(class: &mut glib::Class<Self>) {
+        <Object as IsSubclassable<T>>::override_vfuncs(class);
+
+        let klass = class.as_mut();
+        klass.delete_text = Some(entry_buffer_delete_text::<T>);
+        klass.deleted_text = Some(entry_buffer_deleted_text::<T>);
+        klass.get_length = Some(entry_buffer_get_length::<T>);
+        klass.get_text = Some(entry_buffer_get_text::<T>);
+        klass.insert_text = Some(entry_buffer_insert_text::<T>);
+        klass.inserted_text = Some(entry_buffer_inserted_text::<T>);
+    }
+}
+
+unsafe extern "C" fn entry_buffer_delete_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    n_chars: u32,
+) -> u32 {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.delete_text(wrap.unsafe_cast_ref(), position, n_chars)
+}
+
+unsafe extern "C" fn entry_buffer_deleted_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    n_chars: u32,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.deleted_text(wrap.unsafe_cast_ref(), position, n_chars)
+}
+
+unsafe extern "C" fn entry_buffer_get_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    _n_bytes: *mut usize,
+) -> *const libc::c_char {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.get_text(wrap.unsafe_cast_ref()).to_glib_full()
+}
+
+unsafe extern "C" fn entry_buffer_get_length<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+) -> u32 {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.get_length(wrap.unsafe_cast_ref())
+}
+
+unsafe extern "C" fn entry_buffer_insert_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    charsptr: *const libc::c_char,
+    n_chars: u32,
+) -> u32 {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+    let chars: Borrowed<GString> = from_glib_borrow(charsptr);
+
+    imp.insert_text(
+        wrap.unsafe_cast_ref(),
+        position,
+        &chars[0..(n_chars as usize)],
```

Comment:
Indexing `&str` works in bytes, not characters. You need to use the unicode APIs here.

Reply:
It's in chars, the last push was to be able to switch branches for #94. I will get back to this one once the glib one is in